src/t_mockiato.rs

```diff
@@ -0,0 +1,367 @@
+// vim: tw=80
+/// ```
+/// use mockiato::*;
+///
+/// #[mockable]
+/// pub trait A {
+///     fn foo(&self, key: i16) -> i32;
+/// }
+///
+/// let mut mock = AMock::new();
+/// mock.expect_foo(|k| k.any()).returns(42);
+/// assert_eq!(42, mock.foo(-1));
+/// ```
+fn doctest() {}
+
+#[cfg(test)]
+mod t {
+
+use crate::{TestSuite, UniquelyOwned};
+use mockiato::*;
+
+struct Mockiato {}
+impl TestSuite for Mockiato {
+    const NAME: &'static str = "mockiato";
+
+    fn associated_types() {
+        // Traits are only allowed to contain methods
+        unimplemented!()
+    }
+
+    fn checkpoint() {
+        unimplemented!()
+    }
+
+    fn reference_parameters() {
+        #[mockable]
+        pub trait A {
+            fn foo(&self, x: &u32);
+        }
+
+        let mut mock = AMock::new();
+        mock.expect_foo(|x| x.partial_eq_owned(42)).returns(());
+
+        mock.foo(&42);
+    }
+
+    fn consume_parameters() {
+        // Mockiato can't pass any arguments, by clone or by move, to a return
+        // function, because it doesn't support return functions
+        unimplemented!()
+    }
+
+    fn consume_self() {
+        #[mockable]
+        pub trait A {
+            fn into_nothing(self);
+        }
+
+        let mut mock = AMock::new();
+        mock.expect_into_nothing();
+        mock.into_nothing();
+    }
+
+    fn external_trait() {
```

Comment:
This is possible using the [`remote`](https://docs.rs/mockiato/0.9.0/mockiato/macro.mockable.html#remote) parameter on the `#[mockable]` attribute:

```rust
#[cfg(test)]
#[mockable(remote = "io::Write")]
trait Write {
    fn write(&mut self, buf: &[u8]) -> io::Result<usize>;

    fn flush(&mut self) -> io::Result<()>;

    // Methods with a default implementation can be omitted.
}
```

Reply:
Go it.